src/app/GitCommands/RevisionReader.cs

```diff
@@ -170,38 +170,43 @@ public IReadOnlyCollection<GitRevision> GetRevisionsFromRange(string olderCommit
 
             // output can be cached if Git Notes is not included and hash is valid
             bool canCache = !hasNotes && !string.IsNullOrWhiteSpace(commitHash);
-            if (canCache && GitModule.GitCommandCache.TryGet(arguments.ToString(), out byte[]? commandOutput, out _) is true)
+            ReadOnlyMemory<byte> commandBytes;
+            if (canCache && GitModule.GitCommandCache.TryGet(arguments.ToString(), out string? commandOutput, out _) is true)
             {
-                // OK
+                commandBytes = new ReadOnlyMemory<byte>(Convert.FromBase64String(commandOutput));
             }
             else
             {
 #if DEBUG
                 Debug.WriteLine($"git {arguments}");
 #endif
                 using IProcess process = _module.GitCommandRunner.RunDetached(cancellationToken, arguments, redirectOutput: true, outputEncoding: null);
-                commandOutput = process.StandardOutput.BaseStream.SplitLogOutput().SingleOrDefault().ToArray();
+                commandBytes = new ReadOnlyMemory<byte>(process.StandardOutput.BaseStream.SplitLogOutput().SingleOrDefault().ToArray());
```

Comment:
Should this line be moved down to 193? It doesn't appear to be used until then.

Reply:
OK. Can also move the assign to commandOutput to the end as MRU cache does not need to be updated after reading.